cli/azd/pkg/tools/kubectl/kubectl.go

```diff
@@ -0,0 +1,265 @@
+package kubectl
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/drone/envsubst"
+)
+
+// Executes commands against the Kubernetes CLI
+type KubectlCli interface {
+	tools.ExternalTool
+	// Sets the current working directory
+	Cwd(cwd string)
+	// Sets the env vars available to the CLI
+	SetEnv(env map[string]string)
+	// Applies one or more files from the specified path
+	ApplyFiles(ctx context.Context, path string, flags *KubeCliFlags) error
+	// Applies manifests from the specified input
+	ApplyPipe(ctx context.Context, input string, flags *KubeCliFlags) (*exec.RunResult, error)
+	// Views the current k8s configuration including available clusters, contexts & users
+	ConfigView(ctx context.Context, merge bool, flatten bool, flags *KubeCliFlags) (*exec.RunResult, error)
+	// Sets the k8s context to use for future CLI commands
+	ConfigUseContext(ctx context.Context, name string, flags *KubeCliFlags) (*exec.RunResult, error)
+	// Creates a new k8s namespace with the specified name
+	CreateNamespace(ctx context.Context, name string, flags *KubeCliFlags) (*exec.RunResult, error)
+	// Creates a new generic secret from the specified secret pairs
+	CreateSecretGenericFromLiterals(
+		ctx context.Context,
+		name string,
+		secrets []string,
+		flags *KubeCliFlags,
+	) (*exec.RunResult, error)
+	// Executes a k8s CLI command from the specified arguments and flags
+	Exec(ctx context.Context, flags *KubeCliFlags, args ...string) (exec.RunResult, error)
+}
+
+// K8s CLI Fags
+type KubeCliFlags struct {
+	// The namespace to filter the command or create resources
+	Namespace string
+	// The dry-run type, defaults to empty
+	DryRun string
+	// The expected output, typically JSON or YAML
+	Output string
+}
+
+type kubectlCli struct {
+	tools.ExternalTool
+	commandRunner exec.CommandRunner
+	env           map[string]string
+	cwd           string
+}
+
+// Creates a new K8s CLI instance
+func NewKubectl(commandRunner exec.CommandRunner) KubectlCli {
+	return &kubectlCli{
+		commandRunner: commandRunner,
+	}
+}
+
+// Checks whether or not the K8s CLI is installed and available within the PATH
+func (cli *kubectlCli) CheckInstalled(ctx context.Context) (bool, error) {
+	return tools.ToolInPath("kubectl")
```

Comment:
The `kubectl` is built with go ( yes, cobra again ): https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/kubectl

We should do the same as we do with `bicep` and `ghcli` and pull the binary.  Then customers don't have to worry about the installation process

Reply:
Sounds like an idea for later :)